xwiki-platform-core/xwiki-platform-ratings/xwiki-platform-ratings-ui/src/main/resources/XWiki/Ratings.xml

```diff
@@ -394,25 +394,27 @@ position: relative;
     <className>XWiki.UIExtensionClass</className>
     <guid>d2420ff0-a5b9-4fae-a798-fc623aac7a6b</guid>
     <property>
-      <content>{{include reference='XWiki.RatingsMacros'/}}
-
-{{velocity}}
-#set($configClass     = 'XWiki.RatingsConfigClass')
-#set($globalConfigDoc = $xwiki.getDocument('XWiki.RatingsConfig'))
-#set($webPrefDoc      = $xwiki.getDocument("${doc.space}.WebPreferences"))
-##
-#if($xwiki.exists($webPrefDoc) &amp;&amp; $webPrefDoc.getObjectNumbers($configClass) &gt; 0)
-  #set($configObj = $webPrefDoc.getObject($configClass))
-#else
-  #set($configObj = $globalConfigDoc.getObject($configClass))
-#end
-##
-#set($displayRatings = $configObj.getProperty('displayRatings').value == 1)
-#set($excludedPages  = $stringtool.split($configObj.getProperty('excludedPages').value, ','))
-#set($excludedPages  = $stringtool.stripAll($excludedPages))
-##
-#if($displayRatings &amp;&amp; !$excludedPages.contains($doc.name))
-  #displayFullRating($doc.getDocumentReference() "")
+      <content>{{velocity}}
+#set($canView = $xwiki.getDocument('XWiki.RatingsMacros').hasAccessLevel('view', $xcontext.user))
+#if($canView)
+  {{include reference='XWiki.RatingsMacros'/}}
```

Comment:
The wiki syntax inside the ``{{velocity}}`` macro is evaluated after the Velocity code (because the Velocity code can generate wiki syntax) so the ``{{include}}`` macro (which is wiki syntax) is evaluated (in this case) after ``#displayFullRating()`` is called. Are you sure it works?